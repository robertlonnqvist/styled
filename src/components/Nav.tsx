import { JSX } from "react";
import { Link } from "react-router-dom";
import { routes } from "./routes";

const hrefs = [
  {
    label: "Code",
    href: "https://github.com/robertlonnqvist",
  },
  {
    label: "Contact",
    href: "https://www.linkedin.com/in/robertlonnqvist",
  },
] as const;

const Nav = ({ title }: { title: string }): JSX.Element => {
  return (
    <header className="flex justify-between flex-col md:flex-row items-center mb-6 border-b border-gray-400">
      <a href="/" className="font-extrabold text-2xl hover:text-white mb-3">
        <span className="text-pink-600 ">&gt;&gt;</span> Robert LÃ¶nnqvist
      </a>
      <nav className="flex justify-around space-x-2 items-center mb-3">
        {routes.map(({ label, to }, i) => (
          <Link
            key={i}
            to={to}
            className={`font-bold block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg ${
              title === label ? "bg-gray-500 text-white" : ""
            }`}
          >
            {label}
          </Link>
        ))}
        {hrefs.map(({ label, href }, i) => (
          <a
            key={i}
            href={href}
            className="font-bold block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg"
          >
            {label}
          </a>
        ))}
      </nav>
    </header>
  );
};

export default Nav;
