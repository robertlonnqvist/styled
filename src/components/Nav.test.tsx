import { describe, expect, test } from "@jest/globals";
import { render } from "@testing-library/react";

import Nav from "./Nav";

describe("Nav component", () => {
  test("renders Nav component", () => {
    const { container } = render(<Nav title="About" />);
    expect(container).toMatchSnapshot();
  });
});
