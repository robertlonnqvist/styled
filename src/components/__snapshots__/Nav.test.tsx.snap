// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders Nav component 1`] = `
<div>
  <header
    class="flex justify-between flex-col md:flex-row items-center mb-6 border-b border-gray-400"
  >
    <a
      class="font-extrabold text-2xl hover:text-white mb-3"
      href="/"
    >
      <span
        class="text-pink-600 "
      >
        &gt;&gt;
      </span>
       Robert LÃ¶nnqvist
    </a>
    <nav
      class="flex justify-around space-x-2 items-center mb-3"
    >
      <a
        class="font-bold block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg bg-gray-500 text-white"
        href="/"
      >
        About
      </a>
      <a
        class="font-bold block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg "
        href="/resume"
      >
        Resume
      </a>
      <a
        class="font-bold block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg"
        href="https://github.com/robertlonnqvist"
      >
        Code
      </a>
      <a
        class="font-bold block px-4 py-2 hover:text-white hover:bg-gray-500 rounded-lg"
        href="https://www.linkedin.com/in/robertlonnqvist"
      >
        Contact
      </a>
    </nav>
  </header>
</div>
`;
